machine-learning/tutorials/PhoneCallsAnomalyDetection/Program.cs

```diff
@@ -0,0 +1,107 @@
+
+            //STEP 1: Common data loading configuration
+            // <SnippetLoadData>
+            IDataView dataView = mlContext.Data.LoadFromTextFile<PhoneCallsData>(path: _dataPath, hasHeader: true, separatorChar: ',');
+            // </SnippetLoadData>
+
+            // Detect seasonality for the series
+            // <SnippetCallDetectPeriod>
+            int period = DetectPeriod(mlContext, dataView);
+            // </SnippetCallDetectPeriod>
+
+            // Detect anomaly for the series with period information
+            // <SnippetCallDetectAnomaly>
+            DetectAnomaly(mlContext, dataView, period);
+            // </SnippetCallDetectAnomaly>
+        }
+
+        static int DetectPeriod(MLContext mlContext, IDataView phoneCalls)
+        {
+            Console.WriteLine("Detect period of the series");
+
+            // STEP 2: Detect seasonality
+            // <SnippetDetectSeasonality>
+            int period = mlContext.AnomalyDetection.DetectSeasonality(phoneCalls, nameof(PhoneCallsData.value));
+            // </SnippetDetectSeasonality>
+
+            // <SnippetDisplayPeriod>
+            Console.WriteLine("Period of the series is: {0}.", period);
+            // </SnippetDisplayPeriod>
+
+            return period;
+        }
+
+        static void DetectAnomaly(MLContext mlContext, IDataView phoneCalls, int period)
+        {
+            Console.WriteLine("Detect anomaly points in the series");
+
+            //STEP 2: Setup the parameters
+            // <SnippetSetupSrCnnParameters>
+            var options = new SrCnnEntireAnomalyDetectorOptions()
+            {
+                Threshold = 0.3,
+                Sensitivity = 64.0,
+                DetectMode = SrCnnDetectMode.AnomalyAndMargin,
+                Period = period,
+            };
+            // </SnippetSetupSrCnnParameters>
+
+            //STEP 3: Detect anomaly by SR-CNN algorithm
+            // <SnippetDetectAnomaly>
+            var outputDataView = mlContext.AnomalyDetection.DetectEntireAnomalyBySrCnn(phoneCalls, nameof(PhoneCallsPrediction.Prediction), nameof(PhoneCallsData.value), options);
+            // </SnippetDetectAnomaly>
+
+            // <SnippetCreateEnumerableForResult>
+            var predictions = mlContext.Data.CreateEnumerable<PhoneCallsPrediction>(
+                outputDataView, reuseRowObject: false);
+            // </SnippetCreateEnumerableForResult>
+
+            // <SnippetDisplayHeader>
+            Console.WriteLine("Index\tData\tAnomaly\tAnomalyScore\tMag\tExpectedValue\tBoundaryUnit\tUpperBoundary\tLowerBoundary");
+            // </SnippetDisplayHeader>
+
+            // <SnippetDisplayAnomalyDetectionResults>
+            var index = 0;
+
+            foreach (var p in predictions)
+            {
+                if (p.Prediction[0] == 1)
+                {
+                    Console.WriteLine("{0},{1},{2},{3},{4},{5}  <-- alert is on, detecte anomaly", index,
```

Comment:
This throws an error because there are more placeholders than values. 

`Unhandled exception. System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.`

Look into issue. I got it to work by doing the following: 

```csharp
 Console.WriteLine("{0},{1},{2},{3},{4}  <-- alert is on, detected anomaly", index,
```

Also note had a typo

Reply:
updated.

---
In reply to: [537990262](https://github.com/dotnet/samples/pull/4015#discussion_r537990262) [](ancestors = 537990262)